src/main/java/org/opensrp/web/controller/UniqueIdController.java

```diff
@@ -131,31 +131,30 @@ boolean checkRoleIfRoleExitst(List<String> roleList, String role) {
 	 *
 	 * @return json array object with ids
 	 */
-
+	
 	@RequestMapping(value = "/get", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
-	protected ResponseEntity<String> get(HttpServletRequest request,Authentication authentication) throws JSONException {
+	protected ResponseEntity<String> get(HttpServletRequest request, Authentication authentication)
+	    throws JsonProcessingException, JSONException {
 		
 		String numberToGenerate = getStringFilter("numberToGenerate", request);
 		String source = getStringFilter("source", request);
 		String usedBy = authentication.getName();
 		Map<String, Object> map = new HashMap<>();
 		IdentifierSource identifierSource = identifierSourceService.findByIdentifier(source);
-		try {
-			if (identifierSource != null) {
-				List<String> identifiers = uniqueIdentifierService.generateIdentifiers(identifierSource,
-						Integer.parseInt(numberToGenerate), usedBy);
-				if (identifiers != null) {
-					map.put("identifiers", identifiers);
-				}
-			} else {
-				map.put("identifiers",
-						openmrsIdService.getOpenMRSIdentifiers(source, numberToGenerate));
-			}
+		
+		if (identifierSource != null) {
+			List<String> identifiers = uniqueIdentifierService.generateIdentifiers(identifierSource,
+			    Integer.parseInt(numberToGenerate), usedBy);
+			map.put("identifiers", identifiers);
+			
 			return new ResponseEntity<>(objectMapper.writeValueAsString(map), HttpStatus.OK);
+			
+		} else {
+			
+			return new ResponseEntity<>(String.format("Unique identifier source %d not found", identifierSource),
```

Comment:
not sure %d as identifier is not a number, I think %s should be used, also can we just return id, or identifier instead of the whole object

Reply:
Good catch, its a typo, this should have been the parameter **_source_**

Reply:
Good catch, its a typo, this was meant to be the parameter source

Reply:
Good catch, its a typo, this was meant to be the parameter **source**